ROCm_Tools/ROCm-Tools.rst

```diff
@@ -464,201 +464,230 @@ References
+-  Print the state of wavefronts that report memory violation or upon
+   executing a ``s_trap 2`` instruction.
+-  Allows SIGINT (``ctrl c``) or SIGTERM (``kill -15``) to print
+   wavefront state of aborted GPU dispatches.
+-  It is enabled on Vega10 GPUs on ROCm1.9.
 
-   * A modified version of gdb-7.11 to support GPU debugging. Note the main ROCm specific files are located in gdb-7.11/gdb with the 	  rocm-* prefix.
-   * The ROCm debug facilities library located in amd/HwDbgFacilities/. This library provides symbol processing for GPU kernels.
+Usage
+*****
 
-Build Steps
-############
- 
-1. Clone the ROCm-GDB repository
+To use the ROCr Debug Agent set the following environment variable:
 
-::
-   
-    git clone https://github.com/RadeonOpenCompute/ROCm-GDB.git
+.. code:: sh
 
-2. The gdb build has been modified with new files and configure settings to enable GPU debugging. The scripts below should be run to 	  compile gdb. The run_configure_rocm.sh script calls the GNU autotools configure with additional parameters. The   	 	    	run_configure_rocm.sh script will create the build directory to build the gdb executable in a out of source manner
+   export HSA_TOOLS_LIB=librocr_debug_agent64.so
 
-::
+This will use the ROCr Debug Agent library installed at
+/opt/rocm/lib/librocr_debug_agent64.so by default since the ROCm
+installation adds /opt/rocm/lib to the system library path. To use a
+different version set the LD_LIBRARY_PATH, for example:
 
-    ./run_configure_rocm.sh debug
+.. code:: sh
 
-3.    The run_configure_rocm.sh script also generates the run_make_rocm.sh which sets environment variables for the Make step
+   export LD_LIBRARY_PATH=/path_to_directory_containing_librocr_debug_agent64.so
 
-::
-   
-   ./run_make_rocm.sh
+To display the machine code instructions of wavefronts, together with
+the source text location, the ROCr Debug Agent using the llvm-objdump
+tool. Ensure that a version that supports AMD GCN GPUs is on your
+’$PATH`. For example, for the ROCm1.9:
 
+.. code:: sh
 
-Running ROCm-GDB
-################
+   export PATH=/opt/rocm/opencl/bin/x86_64/:$PATH
 
-The run_make_rocm.sh script builds the gdb executable which will be located in build/gdb/
+Execute your application.
 
-To run the ROCm debugger, you'd also need to get the ROCm GPU Debug SDK.
+If the application encounters a GPU error it will display the wavefront
+state of the GPU to ``stdout``. Possible error states include:
 
-Before running the rocm debugger, the LD_LIBRARY_PATH should include paths to
+-  The GPU executes a memory instruction that causes a memory violation.
+   This is reported as an XNACK error state.
+-  Queue error.
+-  The GPU executes an ``S_TRAP`` instruction. The ``__builtin_trap()``
+   language builtin can be used to generate a ``S_TRAP``.
+-  A SIGINT (``ctrl c``) or SIGTERM (``kill -15``) signal is sent to the
+   application while executing GPU code. Enabled if environment variable
+   ``ROCM_DEBUG_ENABLE_LINUX_SIGNALS=0``.
```

Comment:
=0 or =1 (or nonzero)?

Reply:
Improved/corrected description of the environment variables. Also a value other than 0 or 1 will give an error message.